### PR TITLE
Bukeao habana dlc 1.5.0 release

### PR DESCRIPTION
**Pull request by bukeao**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: habana/tensorflow/training/docker/2.9/py3/synapseai1.5.0/Dockerfile.hpu [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+ENV KMP_AFFINITY=granularity=fine,compact,1,0
+ENV KMP_BLOCKTIME=1
+ENV KMP_SETTINGS=0
+
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PYTHONUNBUFFERED=1
+ENV PYTHONIOENCODING=UTF-8
+ENV LANG=C.UTF-8
+ENV LC_ALL=C.UTF-8
+ENV TF_VERSION=2.8
```
Comment (shantanutrip) [inline habana/tensorflow/training/docker/2.9/py3/synapseai1.5.0/Dockerfile.hpu:23]: Is this 2.9?

File: test/dlc_tests/container_tests/bin/benchmark/run_tensorflow_maskrcnn_training_performance_hpu [hunk 2]
```diff
@@ -15,6 +15,7 @@ cd ${HOME_DIR}/artifacts/gaudi-test-suite
 git checkout r${GIT_BRANCH}
 python3 -m pip install -r requirements.txt
 python3 -m pip install -r test/benchmark/tf-requirements.txt
+python3 -m pip install pycocotools --upgrade
```
Comment (shantanutrip) [inline test/dlc_tests/container_tests/bin/benchmark/run_tensorflow_maskrcnn_training_performance_hpu:18]: Why is this required? Will it make the test backward incompatible?

File: test/test_utils/__init__.py [hunk 3]
```diff
@@ -59,9 +59,9 @@ def get_ami_id_ssm(region_name, parameter_path):
 PT_GPU_PY3_BENCHMARK_IMAGENET_AMI_US_WEST_2 = "ami-02d9a47bc61a31d43"
 # Since latest driver is not in public DLAMI yet, using a custom one
 NEURON_UBUNTU_18_BASE_DLAMI_US_WEST_2 = get_ami_id_boto3(region_name="us-west-2", ami_name_pattern="Deep Learning Base AMI (Ubuntu 18.04) Version ??.?")
-# Habana Base v1.3 ami
-UBUNTU_18_HPU_DLAMI_US_WEST_2 = "ami-08e564663ef2e761c"
-UBUNTU_18_HPU_DLAMI_US_EAST_1 = "ami-06a0a1e2c90bfc1c8"
+# Habana Base v1.5 ami
```
Comment (shantanutrip) [inline test/test_utils/__init__.py:62]: We will make it backward compatible in the future. Can you please add a comment that mentions that AMI ids for all the other SynAI versions?

File: habana/tensorflow/training/docker/2.9/py3/synapseai1.5.0/Dockerfile.hpu [hunk 4]
```diff
@@ -0,0 +1,120 @@
+FROM vault.habana.ai/gaudi-docker/1.5.0/ubuntu20.04/habanalabs/tensorflow-installer-tf-cpu-2.9.1:latest
```
Comment (shantanutrip) [inline habana/tensorflow/training/docker/2.9/py3/synapseai1.5.0/Dockerfile.hpu:1]: I think we haven't added the buildspec for the older synapseAI (1.4.1) version yet. Can those be added please?

File: habana/tensorflow/training/docker/2.9/py3/synapseai1.5.0/Dockerfile.hpu (hunk 1 above)
Comment (bukeao) [reply habana/tensorflow/training/docker/2.9/py3/synapseai1.5.0/Dockerfile.hpu]: fixed

File: habana/tensorflow/training/docker/2.9/py3/synapseai1.5.0/Dockerfile.hpu (hunk 4 above)
Comment (bukeao) [reply habana/tensorflow/training/docker/2.9/py3/synapseai1.5.0/Dockerfile.hpu:1]: do you mean synapseAI(1.4.1) for TF2.9?

File: test/dlc_tests/container_tests/bin/benchmark/run_tensorflow_maskrcnn_training_performance_hpu (hunk 2 above)
Comment (omrialmog) [reply test/dlc_tests/container_tests/bin/benchmark/run_tensorflow_maskrcnn_training_performance_hpu:18]: Required because of a software update in the maskrcnn dependancies. This allows the test to use proper dependancies for the latest version

File: habana/tensorflow/training/docker/2.9/py3/synapseai1.5.0/Dockerfile.hpu (hunk 4 above)
Comment (shantanutrip) [reply habana/tensorflow/training/docker/2.9/py3/synapseai1.5.0/Dockerfile.hpu:1]: I meant `buildspec-1-10-syai-1-4.yml` for PT and `buildspec-2-8-syai-1-4.yml` for TF. (The copies of the buildspec files that we retain in our repo)

Comment (bukeao) [reply habana/tensorflow/training/docker/2.9/py3/synapseai1.5.0/Dockerfile.hpu:1]: done

File: test/test_utils/__init__.py (hunk 3 above)
Comment (bukeao) [reply test/test_utils/__init__.py:74]: done

File: test/dlc_tests/container_tests/bin/benchmark/run_tensorflow_maskrcnn_training_performance_hpu (hunk 2 above)
Comment (shantanutrip) [reply test/dlc_tests/container_tests/bin/benchmark/run_tensorflow_maskrcnn_training_performance_hpu:18]: What about the older synai versions? Will that work normally for them?

File: test/dlc_tests/sanity/test_pre_release.py [hunk 5]
```diff
@@ -241,12 +241,14 @@ def test_framework_version_cpu(image):
         else:
             if "neuron" in image:
                 assert tag_framework_version in output.stdout.strip()
-            if all(_string in image for _string in ["pytorch", "habana", "synapseai1.4.1"]):
+            if (all(_string in image for _string in ["pytorch", "habana", "synapseai1.4.1"])
```
Comment (shantanutrip) [inline test/dlc_tests/sanity/test_pre_release.py:244]: Why is `synapseai1.4.1"` present in `all`?

File: test/test_utils/__init__.py (hunk 3 above)
Comment (shantanutrip) [reply test/test_utils/__init__.py:74]: Thanks a lot, this is extremely beneficial.

File: dlc_developer_config.toml [hunk 6]
```diff
@@ -12,20 +12,20 @@ neuron_mode = false
...
 # Do remember to revert it back to False before merging any PR (including HABANA dedicated PR)
-habana_mode = false
+habana_mode = true
 # Please only set it to True if you are preparing a TRCOMP related PR
 # Do remember to revert it back to False before merging any PR (including TRCOMP dedicated PR)
 trcomp_mode = false
 # Please only set it to True if you are preparing a Benchmark related PR
 # Do remember to revert it back to False before merging any PR (including Benchmark dedicated PR)
-benchmark_mode = false
+benchmark_mode = true
```
Comment (shantanutrip) [inline dlc_developer_config.toml:21]: I think that benchmark mode has run successfully as well, right?

File: test/dlc_tests/sanity/test_pre_release.py (hunk 5 above)
Comment (bukeao) [reply test/dlc_tests/sanity/test_pre_release.py]: fixed

File: dlc_developer_config.toml (hunk 6 above)
Comment (bukeao) [reply dlc_developer_config.toml:21]: yes

File: dlc_developer_config.toml [hunk 7]
```diff
@@ -12,20 +12,20 @@ neuron_mode = false
 graviton_mode = false
 # Please only set it to True if you are preparing a HABANA related PR
```
Comment (shantanutrip) [inline dlc_developer_config.toml:13]: Lets revert the changes in this file

**Review by shantanutrip — APPROVED** [review]
LGTM. Lets revert temp changes.

**Review by shantanutrip — APPROVED** [review]
